pybop/models/base_model.py

```diff
@@ -204,15 +206,16 @@ def simulateS1(self, inputs, t_eval):
                 calculate_sensitivities=True,
             )
 
-            return (
-                sol[self.signal].data,
-                np.asarray(
-                    [
-                        sol[self.signal].sensitivities[key].toarray()
-                        for key in self.fit_keys
-                    ]
-                ).T,
-            )
+            predictions = [sol[signal].data for signal in self.signal]
+
+            sensitivities = [
+                np.array(
+                    [[sol[signal].sensitivities[key]] for signal in self.signal]
+                ).reshape(len(sol[self.signal[0]].data), self.n_outputs)
+                for key in self.fit_keys
+            ]
+
+            return (np.vstack(predictions).T, np.dstack(sensitivities))
```

Comment:
Are we wanting to return a tuple here?

Reply:
I believe this is what was returned before?

Reply:
Ah, right.
`(np.vstack(predictions).T, np.dstack(sensitivities)) == np.vstack(predictions).T, np.dstack(sensitivities)`

 It might be better to leave off the parentheses for consistency.